lib/recent.go

```diff
@@ -0,0 +1,92 @@
+package lib
+
+import (
+	"encoding/json"
+	"os"
+	"path/filepath"
+	"strings"
+)
+
+type RecentFiles struct {
+	Terraform []string `json:"terraform"`
+	Tofu      []string `json:"tofu"`
+}
+
+func addRecent(requestedVersion string, installPath string, dist string) {
+	if !validVersionFormat(requestedVersion) {
+		logger.Errorf("The version %s is not a valid version string and won't be stored", requestedVersion)
+		return
+	}
+	installLocation := GetInstallLocation(installPath)
+	recentFilePath := filepath.Join(installLocation, recentFile)
+	var recentFileData RecentFiles
+	if CheckFileExist(recentFilePath) {
+		unmarshal(recentFilePath, &recentFileData)
+	}
+	var sliceToCheck []string
+	if dist == distTerraform {
+		sliceToCheck = recentFileData.Terraform
+	} else if dist == distTofu {
+		sliceToCheck = recentFileData.Tofu
+	}
+	for _, v := range sliceToCheck {
+		if v == requestedVersion {
+			// entry already exists. Nothing to do
+			return
+		}
+	}
+	if dist == distTerraform {
+		recentFileData.Terraform = append(recentFileData.Terraform, requestedVersion)
+	} else if dist == distTofu {
+		recentFileData.Tofu = append(recentFileData.Tofu, requestedVersion)
+	}
+	saveFile(recentFileData, recentFilePath)
+}
+
+func getRecentVersions(installPath string, dist string) ([]string, error) {
```

Comment:
`getRecentVersions` previously added the suffix " *recent", for display purposes.
I wasn't keen and addressed this in my PR, but the functionality will be lost with this change

Reply:
added that functionality.